addons/spreadsheet_dashboard/static/tests/dashboard/dashboard_action_test.js

```diff
@@ -208,3 +216,113 @@ QUnit.test("Can delete record tag in the filter by hitting Backspace", async fun
     await keyDown({ key: "Backspace" });
     assert.equal(filter.querySelectorAll(".o_tag").length, 0);
 });
+
+QUnit.test("share dashboard from dashboard view", async function (assert) {
+    const target = getFixture();
+    patchWithCleanup(browser, {
+        navigator: {
+            clipboard: {
+                writeText: (url) => {
+                    assert.step("share url copied");
+                    assert.strictEqual(url, "localhost:8069/share/url/132465");
+                },
+            },
+        },
+    });
+    const def = makeDeferred();
+    await createSpreadsheetDashboard({
+        mockRPC: async function (route, args) {
+            if (args.method === "action_get_share_url") {
+                await def;
+                assert.step("dashboard_shared");
+                assert.strictEqual(args.model, "spreadsheet.dashboard.share");
+                return "localhost:8069/share/url/132465";
+            }
+        },
+    });
+    assert.strictEqual(target.querySelector(".spreadsheet_share_dropdown"), null);
+    await click(target, "i.fa-share-alt");
+    assert.equal(
+        target.querySelector(".spreadsheet_share_dropdown")?.innerText,
+        "Generating sharing link"
+    );
+    def.resolve();
+    await nextTick();
+    assert.verifySteps(["dashboard_shared", "share url copied"]);
+    assert.strictEqual(
+        target.querySelector(".o_field_CopyClipboardChar").innerText,
+        "localhost:8069/share/url/132465"
+    );
+    await click(target, ".fa-clipboard");
+    assert.verifySteps(["share url copied"]);
+});
+
+QUnit.test("Changing filter values will create a new share", async function (assert) {
+    const spreadsheetData = {
+        globalFilters: [
+            {
+                id: "1",
+                type: "date",
+                label: "Date Filter",
+                rangeType: "year",
+                defaultValue: {},
+                defaultsToCurrentPeriod: true,
+                pivotFields: {},
+            },
+        ],
+    };
+    const serverData = getServerData(spreadsheetData);
+    const target = getFixture();
+    let counter = 0;
+    patchWithCleanup(browser, {
+        navigator: {
+            clipboard: {
+                writeText: (url) => {},
+            },
+        },
+    });
+    const def = makeDeferred();
```

Comment:
the deferred here looks useless since you are not testing a behavior which happens at the same time as the RPC call. I think you can remove it